app/src/main/java/net/kuama/documentscanner/domain/PerspectiveTransform.kt

```diff
@@ -23,12 +27,25 @@ class PerspectiveTransform : UseCase<Bitmap, PerspectiveTransform.Params>() {
 
     override suspend fun run(params: Params): Either<Failure, Bitmap> = try {
         val src = Mat()
-        Utils.bitmapToMat(params.bitmap, src)
 
-        val tl = params.corners.corners[0] ?: error("Invalid corners")
-        val tr = params.corners.corners[1] ?: error("Invalid corners")
-        val br = params.corners.corners[2] ?: error("Invalid corners")
-        val bl = params.corners.corners[3] ?: error("Invalid corners")
+        Utils.bitmapToMat(
+            params.bitmap, src
+        )
+
+        val orderedCorners = sortPoints(
+            arrayOf(
+                params.corners.corners[0]!!,
```

Comment:
Avoid `!!`, instead keep an explicit error like it was before 
```suggestion
               params.corners.corners[0] ?: error("Invalid corners"),
```